eth/protocols/eth/handlers.go

```diff
@@ -274,6 +285,62 @@ func ServiceGetReceiptsQuery(chain *core.BlockChain, query GetReceiptsRequest) [
 	return receipts
 }
 
+// serviceGetReceiptsQuery69 assembles the response to a receipt query.
+// It does not send the bloom filters for the receipts
+func serviceGetReceiptsQuery69(chain *core.BlockChain, query GetReceiptsRequest) []rlp.RawValue {
+	// Gather state data until the fetch or network limits is reached
+	var (
+		bytes    int
+		receipts []rlp.RawValue
+	)
+	for lookups, hash := range query {
+		if bytes >= softResponseLimit || len(receipts) >= maxReceiptsServe ||
+			lookups >= 2*maxReceiptsServe {
+			break
+		}
+		// Retrieve the requested block's receipts
+		results := chain.GetRawReceiptsByHash(hash)
+		if results == nil {
+			if header := chain.GetHeaderByHash(hash); header == nil || header.ReceiptHash != types.EmptyRootHash {
+				continue
+			}
+		} else {
+			block := chain.GetBlockByHash(hash)
+			results = transformReceipts(results, block.Transactions())
+		}
+		receipts = append(receipts, results)
+		bytes += len(results)
+	}
+	return receipts
+}
+
+// transformReceipts takes a slice of rlp-encoded receipts, and transactions,
+// and applies the type-encoding on the receipts (for non-legacy receipts).
+// e.g. for non-legacy receipts: receipt-data -> {tx-type || receipt-data}
+func transformReceipts(blockReceipts []byte, txs []*types.Transaction) []byte {
+	var (
+		out   bytes.Buffer
+		enc   = rlp.NewEncoderBuffer(&out)
+		it, _ = rlp.NewListIterator(blockReceipts)
+	)
+	outer := enc.List()
+	for i := 0; it.Next(); i++ {
+		if txs[i].Type() == types.LegacyTxType {
+			enc.Write(it.Value())
+		} else {
+			content, _, _ := rlp.SplitList(it.Value())
+			receiptList := enc.List()
+			enc.Write([]byte{txs[i].Type()})
+			enc.Write(content)
+			enc.ListEnd(receiptList)
+		}
+	}
```

Comment:
Yes much better. But perhaps also this? 
```suggestion
	for i := 0; it.Next(); i++ {
		if txs[i].Type() == types.LegacyTxType {
			enc.Write(it.Value())
			continue
		}
		content, _, _ := rlp.SplitList(it.Value())
		receiptList := enc.List()
		enc.Write([]byte{txs[i].Type()})
		enc.Write(content)
		enc.ListEnd(receiptList)
	}
```